skimage/morphology/_skeletonize_3d.py

```diff
@@ -0,0 +1,50 @@
+from __future__ import division, print_function, absolute_import
+
+import numpy as np
+
+from ._skeletonize_3d_cy import _compute_thin_image
+
+
+def _prepare_image(img_in):
+    """Convert to a binary image, pad the it w/ zeros, and ensure it's 3D.
+    """
+    if img_in.ndim < 2 or img_in.ndim > 3:
+        raise ValueError('expect 2D, got ndim = %s' % img_in.ndim)
+
+    img = img_in.copy()
+    if img.ndim == 2:
+        img = img.reshape((1,) + img.shape)
+
+    intensity = img.max()
+
+    # normalize to binary
+    img[img != 0] = 1
+
+    # pad w/ zeros to simplify dealing w/ neighborhood of a pixel
+    img_o = np.zeros(tuple(s + 2 for s in img.shape),
+                     dtype=np.uint8)
+    img_o[1:-1, 1:-1, 1:-1] = img.astype(np.uint8)
```

Comment:
then also needs `mode="constant", constant_values=0`. Uh, no, it defaults to zero, ok. Too magic to my own taste, but ok, can change.